packages/components/src/local/route/helpers/planned-routes-for.ts

```diff
@@ -0,0 +1,56 @@
+import { LatLng } from 'leaflet'
+
+import SergeHex from '../../mapping/types/serge-hex'
+import SergeGrid from '../../mapping/types/serge-grid'
+import hexNamed from '../../hex-grid/helpers/hex-named'
+
+export interface RouteData {
+  /**
+   *  line representing this route (may include multiple steps per turn)
+   */
+  polyline: LatLng[]
+  /** 
+   * series of end-of-turn locations, one per turn
+   */ 
+  turnEnds: LatLng[]
+}
+
+export const LENGTH_OF_TRIMMED_LINE: number = 2
+
+const plannedRoutesFor = (gridCells: SergeGrid<SergeHex<{}>>, position: string, steps: [any], 
+  trimmed: boolean): RouteData => {
+    const polyline: LatLng[] = []
+    const turnEnds: LatLng[] = []
+    var stepCtr: number = 0
+    // start with current position
+    const startCell: SergeHex<{}> | undefined = hexNamed(position, gridCells)
+    if(startCell) {
+      const startPos: LatLng = startCell.centreLatLng
+      if(steps) {
+        // store the line start
+        polyline.push(startPos)
+        steps.forEach((step:any) => {
```

Comment:
This is VERY similar to the `history-routes-for` file. Is there no way these could be unified into a single function? There is definitely some scope for refactoring here, not all of it will need to be repeated.

Reply:
Yes, I thought of a way of doing it whilst walking.  It's an artificial problem anyway. In our sample data, our past routes are stored in a different way to our future routes.  Once we're through this stage, we should change the data to be more consistent.